### PR TITLE
Substitute traceid for msgid in rpc

### PR DESCRIPTION
**Pull request by smellthemoon**
Signed-off-by: lixinguo <xinguo.li@zilliz.com>
#19639

### CODE REVIEW

File: internal/proxy/task_query.go [hunk 1]
```diff
@@ -169,46 +170,61 @@ func (t *queryTask) PreExecute(ctx context.Context) error {
 
 	collectionName := t.request.CollectionName
 	t.collectionName = collectionName
+	traceID, _, _ := trace.InfoFromContext(ctx)
 	if err := validateCollectionName(collectionName); err != nil {
-		log.Ctx(ctx).Warn("Invalid collection name.", zap.String("collectionName", collectionName),
-			zap.Int64("msgID", t.ID()), zap.String("requestType", "query"))
+		log.Ctx(ctx).Warn("Invalid collection name.",
+			zap.String("collectionName", collectionName),
+			zap.String("traceID", traceID),
```
[the remark this answers is not in the document]
Comment (jiaoew1991) [reply internal/proxy/task_query.go:177]: Other logs in this file have the same problem

[the remark this answers is not in the document]
Comment (smellthemoon) [reply internal/proxy/task_query.go]: ok

File: internal/proxy/task.go [hunk 2]
```diff
@@ -1286,7 +1287,10 @@ func (lct *loadCollectionTask) OnEnqueue() error {
 }
 
 func (lct *loadCollectionTask) PreExecute(ctx context.Context) error {
-	log.Debug("loadCollectionTask PreExecute", zap.String("role", typeutil.ProxyRole), zap.Int64("msgID", lct.Base.MsgID))
+	traceID, _, _ := trace.InfoFromContext(ctx)
+	log.Debug("loadCollectionTask PreExecute",
```
Comment (yah01) [inline internal/proxy/task.go:1291]: log.Ctx(ctx).Debug() may be better, this makes sure all fields in context will be logged, more extensible

File: internal/proxy/task.go [hunk 3]
```diff
@@ -1305,7 +1309,10 @@ func (lct *loadCollectionTask) PreExecute(ctx context.Context) error {
 }
 
 func (lct *loadCollectionTask) Execute(ctx context.Context) (err error) {
-	log.Debug("loadCollectionTask Execute", zap.String("role", typeutil.ProxyRole), zap.Int64("msgID", lct.Base.MsgID))
+	traceID, _, _ := trace.InfoFromContext(ctx)
+	log.Debug("loadCollectionTask Execute",
```
Comment (yah01) [inline internal/proxy/task.go:1313]: use log.With() to make sure all logs here will output role, traceID and collectionID, 1313 misses collectionID

File: internal/proxy/task.go [hunk 4]
```diff
@@ -1365,8 +1374,10 @@ func (lct *loadCollectionTask) Execute(ctx context.Context) (err error) {
 }
 
 func (lct *loadCollectionTask) PostExecute(ctx context.Context) error {
-	log.Debug("loadCollectionTask PostExecute", zap.String("role", typeutil.ProxyRole),
-		zap.Int64("msgID", lct.Base.MsgID))
+	traceID, _, _ := trace.InfoFromContext(ctx)
+	log.Debug("loadCollectionTask PostExecute",
```
Comment (yah01) [inline internal/proxy/task.go:1378]: Need collectionID or collectionName

File: internal/querycoordv2/job/scheduler.go [hunk 5]
```diff
@@ -147,8 +148,9 @@ func (scheduler *Scheduler) processQueue(collection int64, queue jobQueue) {
 }
 
 func (scheduler *Scheduler) process(job Job) {
+	traceID, _, _ := trace.InfoFromContext(job.Context())
 	log := log.With(
```
Comment (yah01) [inline internal/querycoordv2/job/scheduler.go:152]: log.Ctx(ctx).With(), and remove the `zap.String("traceID")`

**Review by yah01 — CHANGES_REQUESTED** [review]

File: internal/proxy/task_insert.go [hunk 6]
```diff
@@ -232,25 +232,29 @@ func (it *insertTask) PreExecute(ctx context.Context) error {
 	// check primaryFieldData whether autoID is true or not
 	// set rowIDs as primary data if autoID == true
 	err = it.checkPrimaryFieldData()
+	log := log.Ctx(ctx).With(zap.String("collection name", collectionName))
```
Comment (yah01) [inline internal/proxy/task_insert.go:235]: log field name convention: camelCase, so `collection name` -> `collectionName`

File: internal/proxy/task_insert.go [hunk 7]
```diff
@@ -443,8 +447,11 @@ func (it *insertTask) Execute(ctx context.Context) error {
 	tr.Record("get used message stream")
 
 	channelNames, err := it.chMgr.getVChannels(collID)
+	traceID, _, _ := trace.InfoFromContext(ctx)
```
Comment (yah01) [inline internal/proxy/task_insert.go:450]: log.Ctx(ctx).With(...), avoid logging traceID and collectionID again and again

File: internal/proxy/task_query.go [hunk 8]
```diff
@@ -170,45 +170,51 @@ func (t *queryTask) PreExecute(ctx context.Context) error {
 	collectionName := t.request.CollectionName
 	t.collectionName = collectionName
 	if err := validateCollectionName(collectionName); err != nil {
-		log.Ctx(ctx).Warn("Invalid collection name.", zap.String("collectionName", collectionName),
-			zap.Int64("msgID", t.ID()), zap.String("requestType", "query"))
+		log.Ctx(ctx).Warn("Invalid collection name.",
```
Comment (yah01) [inline internal/proxy/task_query.go:173]: same as `task_insert.go:450`

File: internal/proxy/task_query.go [hunk 9]
```diff
@@ -314,15 +323,15 @@ func (t *queryTask) Execute(ctx context.Context) error {
 	err := executeQuery(WithCache)
 	if errors.Is(err, errInvalidShardLeaders) || funcutil.IsGrpcErr(err) || errors.Is(err, grpcclient.ErrConnect) {
 		log.Ctx(ctx).Warn("invalid shard leaders cache, updating shardleader caches and retry search",
```
Comment (yah01) [inline internal/proxy/task_query.go:325]: same as task_insert.go:450

File: internal/proxy/task_insert.go (hunk 7 above)
Comment (smellthemoon) [reply internal/proxy/task_insert.go:450]: ok，I will do this in the whole package at the next pr.

**Comment by smellthemoon** [comment]
rerun ut

**Comment by smellthemoon** [comment]
rerun ut

**Comment by smellthemoon** [comment]
rerun ut

**Comment by smellthemoon** [comment]
rerun ut

**Comment by smellthemoon** [comment]
rerun ut

**Pull request merged**